src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -399,21 +399,35 @@ private ISet<DocumentId> GetProjectOpenDocuments_NoLock(ProjectId project)
             DocumentId documentId, SourceTextContainer textContainer,
             bool isCurrentContext = true)
         {
-            OnDocumentOpenedAsync(documentId, textContainer, isCurrentContext).Wait(CancellationToken.None);
+            OnDocumentOpenedAsync(documentId, textContainer, isCurrentContext, wait: true).Wait(CancellationToken.None);
         }
 
-        internal async Task OnDocumentOpenedAsync(
-            DocumentId documentId, SourceTextContainer textContainer, 
+        internal Task OnDocumentOpenedAsync(
+            DocumentId documentId, SourceTextContainer textContainer,
             bool isCurrentContext = true)
         {
+            return OnDocumentOpenedAsync(documentId, textContainer, isCurrentContext, wait: false);
+        }
+
+        private async Task OnDocumentOpenedAsync(
+            DocumentId documentId, SourceTextContainer textContainer, 
+            bool isCurrentContext,
+            bool wait)
```

Comment:
No.  The "fix" here is that the event we fire is marshalled to the how scheduler (just like all other workspace events).  Basically, what Jason and i think is that this was an issue because this one event behaves different from all our other workspace events.  THey can be scheduled on the background, and fire on the FG just fine.  This single event was fired on the calling thread.  

We think that isn't necessary, and it should follow hte normal model.